src/cpp/desktop/DesktopWebProfile.cpp

```diff
@@ -66,9 +66,23 @@ class Interceptor : public QWebEngineUrlRequestInterceptor
 
 } // end anonymous namespace
 
-WebProfile::WebProfile(const QUrl& baseUrl, QObject* parent)
-   : QWebEngineProfile(QString::fromUtf8("rstudio-desktop"), parent)
+WebProfile* WebProfile::getInstance(const QUrl &baseUrl)
 {
+   static WebProfile* instance = new WebProfile(baseUrl);
+   return instance;
```

Comment:
Don't we need to ensure that a different WebProfile object is constructed for different base urls to ensure the interceptor code (for setting the shared secret header) works properly? This seems like that would break that.